WearOAuth/oauth-device-grant/src/main/java/com/example/android/wearable/oauth/devicegrant/AuthDeviceGrantActivity.kt

```diff
@@ -59,42 +60,76 @@ class AuthDeviceGrantActivity : ComponentActivity() {
         val uiState = deviceGrantViewModel.uiState.collectAsState()
-        val localContext = LocalContext.current
-        val columnState = rememberResponsiveColumnState(
-            contentPadding = ScalingLazyColumnDefaults.padding(
-                first = ItemType.Text,
-                last = ItemType.Text
-            )
+        AuthenticateScreen(
+            uiState.value.statusCode,
+            uiState.value.resultMessage,
+            deviceGrantViewModel::startAuthFlow
+        )
+    }
+}
+
+@OptIn(ExperimentalHorologistApi::class)
+@Composable
+fun AuthenticateScreen(
+    statusCode: Int,
+    resultMessage: String,
+    startAuthFlow: () -> Unit
+) {
+    val columnState = rememberResponsiveColumnState(
+        contentPadding = ScalingLazyColumnDefaults.padding(
+            first = ItemType.Text,
+            last = ItemType.Text
         )
-        ScreenScaffold(scrollState = columnState) {
-            ScalingLazyColumn(columnState = columnState) {
-                item {
-                    ListHeader {
-                        Text(
-                            stringResource(R.string.oauth_device_auth_grant),
-                            textAlign = TextAlign.Center
-                        )
-                    }
+    )
+    ScreenScaffold(scrollState = columnState) {
+        ScalingLazyColumn(columnState = columnState) {
+            item {
+                ListHeader {
+                    Text(
+                        stringResource(R.string.oauth_device_auth_grant),
+                        textAlign = TextAlign.Center
+                    )
                 }
-                item {
-                    Button(
-                        onClick = { deviceGrantViewModel.startAuthFlow(localContext) },
-                        modifier = Modifier.fillMaxSize()
-                    ) {
-                        Text(
-                            text = stringResource(R.string.get_grant_from_phone),
-                            modifier = Modifier.align(Alignment.CenterVertically)
-                        )
-                    }
+            }
+            item {
+                Button(
+                    onClick = { startAuthFlow() },
+                    modifier = Modifier.fillMaxSize()
+                ) {
+                    Text(
+                        text = stringResource(R.string.get_grant_from_phone),
+                        modifier = Modifier.align(Alignment.CenterVertically)
+                    )
                 }
-                item { Text(uiState.value.statusCode.toString()) }
-                item { Text(uiState.value.resultMessage) }
             }
+            item { Text(stringResource(id = statusCode)) }
+            item { Text(resultMessage) }
         }
     }
 }
+
+@WearPreviewDevices
+@WearPreviewFontScales
+@Composable
+fun AuthenticateScreenPreview() {
+    AuthenticateScreen(
+        statusCode = R.string.status_retrieved,
+        resultMessage = "User name",
```

Comment:
nit: For future, "User name" isn't the most exciting name here in screenshots.

Maybe something with more character like a "Bobby Bonson" or a "Randle McMurphy"?